store/store.go

```diff
@@ -2515,3 +2518,19 @@ func (s *Store) CreateCohorts(ctx context.Context, snaps []string) (map[string]s
 
 	return remote.CohortKeys, nil
 }
+
+type userAgentContextKey struct{}
+
+// ClientUserAgentContext carries the client user agent that talks to snapd
+func ClientUserAgentContext(ua string) context.Context {
```

Comment:
not a blocker, but I need to think about the name/signature of this

Reply:
I would prefer a signature:

    WithClientUserAgent(parent Context,  req *http.Request)  context.Context

and forcing daemon to use `Background` or `TODO`